### DISCUSSION TITLE
sampling monotonic functions from ```MultivariateNormal```

### DISCUSSION THREAD

**Discussion by AndreaBraschi** [cornellius-gp/gpytorch#2329] [Q&A]
Hi all,

I would like to sample monotonic functions from a ```MultivariateNormal```. So far, I've tried to construct the distribution with a monotonic mean, but I havent' been succesful so far. Here's the code that I've written:

```
torch.manual_seed(123)
x = torch.linspace(0, 10, 100, requires_grad=True).unsqueeze(-1)
num_tasks = 10
list = []
torch.manual_seed(3)
kernel = gpytorch.kernels.ScaleKernel(gpytorch.kernels.RBFKernel())
for i in range(num_tasks):
    K = kernel(x).add_jitter(1e-4)
    L = psd_safe_cholesky(K.evaluate())
    dist = MultivariateNormal(x[:, 0], CholLinearOperator(TriangularLinearOperator(L)))
    sample = dist.sample(torch.Size([1]))
    list.append(sample.detach().numpy())
torch.manual_seed(torch.seed())

samples = torch.Tensor(list).squeeze(1)

```
any suggestion will be greatly appreciated.

**Comment by Balandat** [comment]
> I would like to sample monotonic functions from a MultivariateNormal

I am not sure I understand this statement, can you elaborate? What function are you sampling? Are you trying to draw GP sample paths?

**Comment by AndreaBraschi** [comment]
I'm trying to draw monotonic samples, sorry.

**Reply by Balandat** [reply]
Still unclear. Are you trying to draw samples on a sequence of successively truncated MVNs?

**Comment by AndreaBraschi** [comment]
Not really. I was just trying to understand if there is a way to construct a ```MultivariateNormal``` from which monotonic samples can be drawn. I thought using a monotonic mean could be a way of doing it, but I was wrong. Maybe should I build a ```kernel``` that enforces monotonicity in the samples?

**Reply by gpleiss** [reply]
To the best of my knowledge, there are no kernels that yield monotonicity, except for a LinearKernel.
